src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -217,6 +215,14 @@ public void RunAndBlock(CancellationToken cancellationToken = default(Cancellati
                 }
                 catch (Exception ex)
                 {
+                    if (_disposed)
+                    {
+                        // In some cases during shutdown we'll be disposed and get
+                        // some terminating exceptions. We want to just ignore these
+                        // and stop immediately.
```

Comment:
Reproducing this locally during web server shutdown, I'd get a ThreadAbort exception here. Everything was tearing down, we've already been disposed. In such cases we don't want to write logs or process these errors.